### PR TITLE
Ismail Kuplu - Week 2 (added JavaScript exercise files and the Grade calculator project within a homework file)

### PR DESCRIPTION
**Pull request by ikuplu**

### CODE REVIEW

File: Week2/homework/js-exercises/booksArray.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+]
+
+// Loop through the array of books.
+for (let i in booksArray) {
+
+  // For each book, log the book title and book author.
+  console.log('"' + booksArray[i].title + ' by ' + booksArray[i].author + '"');
+
+  // Create a conditional statement to change the log depending on whether you read it yet or not.
+  if (booksArray[i].alreadyRead === true) {
```
Comment (bnhoogendijk) [inline Week2/homework/js-exercises/booksArray.js:31]: Could you think of an even simpler way to express this condition? Note that the `alreadyRead` property is boolean.

File: Week2/homework/js-exercises/booksArray.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ },
+ {
+  'title': 'The Chronicles of Narnia',
+  'author': 'C. S. Lewis',
+  alreadyRead: false,
+ }
+]
+
+// Loop through the array of books.
+for (let i in booksArray) {
```
Comment (bnhoogendijk) [inline Week2/homework/js-exercises/booksArray.js:25]: Excellent! This allows you to loop over the items of the array by index. Two other ways are to use a `for (let item of booksArray)` loop, or an array function `booksArray.forEach(callback)` (this latter will be discussed in week 3 if I am not mistaken.) Choose your own favorite 😉

File: Week2/homework/js-exercises/getDrinks.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+/*Create a loop that runs 5 times. On each iteration, push a drink into the drinkTray variable. 
+The drinkTray can only hold at most two instances of the same drink type. */
+let i = 0;
+while (i < drinkTypes.length) {
+  drinkTray.push(drinkTypes[i]);
+
+  if (drinkTray.length === 5) {
+    break;
+  }
```
Comment (bnhoogendijk) [inline Week2/homework/js-exercises/getDrinks.js:15]: You made it work! The output of the function is right. You could consider a closer look at you loop. Is the condition of your while loop (`I < drinkTypes.lengt`) ever exceeded? The loop actually breaks on the `if-statement`. This is not wrong, but this might be a bit difficult to interpret. Could you think of a way to replace the while condition and leave out the if statement?

File: Week2/homework/js-exercises/recipeList.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+/*Give the empty object 3 properties: a title (string), a servings (number) and 
+an ingredients (array of strings) property.*/
+myRecipe.title = 'Pilav';
+myRecipe.servings = 4;
+myRecipe.ingredients = ['400 grams rice', ' 2 tablespoon butter', ' 3 glass water', ' 1/4 tablespoon salt'];
+
+
+// Log each property out separately, using a loop.
+for (let key in myRecipe) {
+  if (myRecipe.hasOwnProperty(key)) {
```
Comment (bnhoogendijk) [inline Week2/homework/js-exercises/recipeList.js:22]: What is you intention with this condition? Please consider this example and assess whether or not you achieved wat you intended:
```
o = new Object();
o.propOne = null;
o.hasOwnProperty('propOne');   // returns true
o.propTwo = undefined;  
o.hasOwnProperty('propTwo');   // returns true
```

File: Week2/homework/js-exercises/recipeList.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  console.log(myRecipeKeys[key] + ": " +  myRecipe[key]);
```
Comment (bnhoogendijk) [inline Week2/homework/js-exercises/recipeList.js:23]: Apart from the previous comment you still did an excellent job, making this exercise work and made an extra effort to make the labels more readable and pretty!

File: Week2/homework/js-exercises/removeComma.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+// Declare a string variable called myString.
+let myString = 'hello,this,is,a,difficult,to,read,sentence';
+
+// Log the length of myString.
+console.log(myString.length);
+
+/* Replace the commas with spaces by using replace method to make it more readable. */
+myString = myString.replace(/,/g, " ");
```
Comment (bnhoogendijk) [inline Week2/homework/js-exercises/removeComma.js:10]: Excellent use of the regular expression and global flag!

File: Week2/homework/js-exercises/reportEvenOdd.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+for (let i = 0; i <= 20; i++) {
+
+  /* Create a conditional statement that checks if the value of the counter variable is odd or even,
+     log the result to the console. */
+  if (i % 2 === 0) {
+    console.log("The number " + i + " is even!");
+  } else {
+    console.log("The number " + i + " is odd!");
+  }
+}
```
Comment (bnhoogendijk) [inline Week2/homework/js-exercises/reportEvenOdd.js:13]: Woop, woop! Perfect!

**Review by bnhoogendijk — APPROVED** [review]
Very well done on this weeks homework 👏 ! You demonstrated good understanding of loops and conditions! I left you some comments to think about, but don't feel the need to commit extra changes!

File: Week2/homework/js-exercises/getDrinks.js (hunk 3 above)
Comment (ikuplu) [reply Week2/homework/js-exercises/getDrinks.js:15]: How about this as a shorter way?

`for (let i = 0; i < 5; i++) {
  drinkTray.push(Math.floor(drinkTypes[i/2]));
}`

File: Week2/homework/calculateGrade.js [hunk 8]
```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+// Write a function.
+function calculateGrade(testScore) {
+  
+
+  // Converts the score into a percentage.
+  let percentage = Math.floor((testScore / 100) * 100);
```
[the remark this answers is not in the document]
Comment (ikuplu) [reply Week2/homework/calculateGrade.js:8]: Actually it adds nothing that's right. I took that depending on the example given in the exercise. Now I have updated the code with a formula computing the percentage for any score range, as follows: 

`// Write a function.
function calculateGrade(testScore, scoreMax, scoreMin) {
  

  // Converts the score into a percentage.
  let percentage = Math.floor((testScore - scoreMin) / (scoreMax - scoreMin) * 100);`

File: Week2/homework/calculateGrade.js [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+      break;
+    case 0 <= percentage && percentage <= 49:
+      grade = "F";
+      break;
+    default:
+      break;
+  }
+
+  // Returns the result to the console with a template string.
+  console.log("You got a " + grade + " (" + percentage + "%)!");
```
[the remark this answers is not in the document]
Comment (ikuplu) [reply Week2/homework/calculateGrade.js:36]: I've changed the log sentence with a template string as follows:

`console.log(`You got a ${grade} (${percentage}%)!`);`

File: Week2/homework/js-exercises/booksArray.js (hunk 1 above)
Comment (ikuplu) [reply Week2/homework/js-exercises/booksArray.js:31]: I've updated with a shorter form as;

`if (booksArray[i].alreadyRead) {`

File: Week2/homework/js-exercises/booksArray.js (hunk 2 above)
Comment (ikuplu) [reply Week2/homework/js-exercises/booksArray.js:25]: I'll keep this in mind, seem to go more commonly with forEach in JS, thanks ;)

File: Week2/homework/js-exercises/recipeList.js (hunk 4 above)
Comment (ikuplu) [reply Week2/homework/js-exercises/recipeList.js:22]: I put that to check if the object has that key inside, but I see now that it's really unnecessary. I've taken that if condition out.

File: Week2/homework/js-exercises/recipeList.js (hunk 5 above)
Comment (ikuplu) [reply Week2/homework/js-exercises/recipeList.js:23]: Thanks very much!

File: Week2/homework/js-exercises/removeComma.js (hunk 6 above)
Comment (ikuplu) [reply Week2/homework/js-exercises/removeComma.js:10]: Thanks :)

File: Week2/homework/js-exercises/reportEvenOdd.js (hunk 7 above)
Comment (ikuplu) [reply Week2/homework/js-exercises/reportEvenOdd.js:13]: Thanks! :)

File: Week2/homework/calculateGrade.js [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+
+// Test some corner values.
+calculateGrade(0);
+calculateGrade(49);
+calculateGrade(50);
+calculateGrade(69)
+calculateGrade(70);
+calculateGrade(89);
+calculateGrade(90);
+calculateGrade(100);
```
[the remark this answers is not in the document]
Comment (ikuplu) [reply Week2/homework/calculateGrade.js:48]: Thanks very much Bas! 👍

**Comment by ikuplu** [comment]
Thank you very much for your effort and time for the feedback Bas, very helpful for me, I appreciate it. I've implemented all and committed in my local environment. But don't push them again to your side as you've written no need to, instead I've put copies of the updates inside the 'reply' fields. 👍

File: Week2/homework/js-exercises/getDrinks.js (hunk 3 above)
Comment (bnhoogendijk) [reply Week2/homework/js-exercises/getDrinks.js:15]: Great!
